src/java/org/apache/cassandra/repair/RepairCoordinator.java

```diff
@@ -447,11 +452,12 @@ else if (isMeta && !isCMS)
 
         if (shouldExcludeDeadParticipants)
         {
-            Set<InetAddressAndPort> actualNeighbors = Sets.newHashSet(Iterables.filter(allNeighbors, ctx.failureDetector()::isAlive));
-            shouldExcludeDeadParticipants = !allNeighbors.equals(actualNeighbors);
-            allNeighbors = actualNeighbors;
+            Set<InetAddressAndPort> actualNeighbors = Sets.newHashSet(Iterables.filter(includeNeighbors, ctx.failureDetector()::isAlive));
+            shouldExcludeDeadParticipants = !includeNeighbors.equals(actualNeighbors);
+            if (shouldExcludeDeadParticipants) includeNeighbors = actualNeighbors;
+            else logger.info("{} all replicas {} considered up and healthy; clearing force flag for this job", state.id, includeNeighbors);
         }
-        return new NeighborsAndRanges(shouldExcludeDeadParticipants, allNeighbors, commonRanges);
+        return new NeighborsAndRanges(shouldExcludeDeadParticipants, includeNeighbors.containsAll(allNeighbors), includeNeighbors, commonRanges);
```

Comment:
Shouldn't it be `allNeighbors.containsAll(includeNeighbors)`?

Reply:
hm, actually I think it doesn't matter in this case, but I think we can avoid building `allNeighbors`, just setting `false` if we `continue`, or?